### DISCUSSION TITLE
simple theme - Hero image behaves different to hero text

### DISCUSSION THREAD

**Discussion by stzi** [GetPublii/Publii#1669] [Bug Report]
### Operating system

macOS 14.6.1 (23G93)

### Publii version

0.46.1 (build 16918)

### Issue type

Free theme

### Bug description

The hero image scales nicely with the width of the site. It is exactly as wide as the hero text. This changes at the point, where the text reaches its max. witdth. Now the image keeps growing while the page is stretched further. I guess this not intended behavior as it (a) looks weird and (b) scales the image further up in the background leading to a zoom effect.

### Steps to reproduce

Add image to hero section and stretch the width of the browser window, I am on a 5K iMac and it starts to look annoying when about two thirds of the screen width is used.

**Comment by GetPublii** [comment]
`I guess this not intended behavior`

It is. The image is set to cover in CSS, meaning it's stretched to the specified height of the hero section. This prevents it from scaling indefinitely, which could result in the image occupying one or more screen heights on larger displays like 5K. To avoid this, the height needs to be constrained, and if the image isn't set to cover, it will stretch unnaturally in width. This approach to handling images is quite common. You can modify this behavior via CSS, but it's unlikely to look better than it does now.

**Reply by bobmitro** [reply]
Keep in mind that no one stretches the browser across devices. Responsiveness is about adapting content to different screen sizes, not stretching the browser window. Stretching might be done for testing purposes.

**Reply by stzi** [reply]
You are right I guess and my point is mostly academic. On my MacBook Air it looks quite good. But I am still not convinced that the behavior is optimal. If I fiddle around in the generated code I can achieve what I think is right by moving the hero image into the same container that holds the

's with the text. Then the image scales in width with the text and stops getting wider when then text does. looks perfekt. Unfortunately I could not figure out how to realize this in custom CSS (not a web developer). I can accept if the opinion is "good enough", maybe my Idea would break other stuff? But in the end it does not look good if the image is broader than the text above if one scales it beyond a certain size.
